app/models/rbac.rb

```diff
@@ -248,7 +248,33 @@ def self.find_targets_with_user_group_rbac(klass, scope, rbac_filters, find_opti
     end
   end
 
+  def self.group(user_or_group)
+    case user_or_group
+    when User
+      user_or_group.current_group
+    when MiqGroup
+      user_or_group
+    when NilClass
+    else
+      raise
+    end
+  end
+
+  def self.find_options_for_tenant(klass, user_or_group, find_options = {})
+    tenant_id = group(user_or_group).try(:tenant_id)
+    return find_options unless tenant_id
+
+    tenant_id_clause = ["#{klass.table_name}.tenant_id = ? OR #{klass.table_name}.tenant_id IS NULL", tenant_id]
```

Comment:
Great, yeah, see my diagram above, not sure about NULL, I added it to spark conversation.